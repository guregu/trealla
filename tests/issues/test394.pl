:- initialization(main).
:- use_module(library(dif)).

ti(G=Rs) :-
   ti(EsG, EsRG, 3),
   ( G = EsG ; G = EsRG ),
   findall(R,(call_residue_vars(G,Vs),length(Vs,R)),Rs).

ti(EsG,(A,B,EDif),N) :-
   N>0,
   EDif = dif(_,_),
   EsG = (EDif,A,B),
   f(EsG).

f((dif(A,B),B=[]*[],A=[]*_)).

main :-
	findall(G-Rs, ti(G=Rs), L),
	write(L), nl.

