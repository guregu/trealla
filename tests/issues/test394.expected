[(dif(_22,_23),_23=[]*[],_22=[]*_24)-[1],(_25=[]*[],_26=[]*_27,dif(_26,_25))-[1]]
